Hmac.js

```diff
@@ -48,12 +52,24 @@ export class Hmac {
    *
    * @param {object} options - The options to use.
    * @param {Uint8Array} options.data - The data to sign as a Uint8Array.
+   * @param {boolean} options.useCache - Enable the use of a cache.
    *
    * @returns {Promise<string>} The base64url-encoded signature.
    */
-  async sign({data}) {
+  async sign({data, useCache = true}) {
+    if(useCache) {
+      const sig = this.cache.get(base64url.encode(data));
+      if(sig) {
+        return sig;
+      }
+    }
     const {id: keyId, kmsClient, capability, invocationSigner} = this;
-    return kmsClient.sign({keyId, data, capability, invocationSigner});
+    const sig = await kmsClient.sign(
+      {keyId, data, capability, invocationSigner});
```

Comment:
The cachability of the result just depends on `data`?  These other params won't change the result?  If so, the key would have to be based on all the params.

Reply:
@davidlehn The other params are member fields defined during the instantiation of the object. I am not aware of any use case for changing those member fields. With that assumption, I made the decision that they were static data during the lifecycle of the object. This assumption could very well be wrong. Is there something that you can point me to that suggests that there are cases where one is modifying those member variables on the same object?

Reply:
I'm not familiar enough with the use case.  If those values don't change between calls here then this is fine.